digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -181,10 +181,37 @@ public TaskResult runTask()
             AWSCredentials credentials = credentials(tag);
 
             SecretProvider awsSecrets = context.getSecrets().getSecrets("aws");
+            SecretProvider s3Secrets = awsSecrets.getSecrets("s3");
+            SecretProvider emrSecrets = awsSecrets.getSecrets("emr");
+            SecretProvider kmsSecrets = awsSecrets.getSecrets("kms");
+
+            Optional<String> emrEndpoint = Aws.first(
+                    () -> emrSecrets.getSecretOptional("endpoint"),
+                    () -> params.getOptional("endpoint", String.class));
```

Comment:
Oh, I see. I was a bit confused since `params.getOptional("endpoint")` was used for `kmsEndpoint`.
Why did you use `params.getOptional("endpoint")` only for `kmsEndpoint`? It sounds a bit inconsistent behaviour to me.

Also, CMIIW a common `params.getOptional("endpoint")` doesn't work over the three services, right?